Core/src/Geometry/CuboidVolumeBuilder.cpp

```diff
@@ -61,32 +60,63 @@ std::shared_ptr<const Acts::Layer> Acts::CuboidVolumeBuilder::buildLayer(
     const GeometryContext& gctx,
     Acts::CuboidVolumeBuilder::LayerConfig& cfg) const {
   // Build the surface
-  if (cfg.surface == nullptr) {
-    cfg.surface = buildSurface(gctx, cfg.surfaceCfg);
+  if (cfg.surfaces.empty()) {
+    for (const auto& sCfg : cfg.surfaceCfg) {
+      cfg.surfaces.push_back(buildSurface(gctx, sCfg));
+    }
   }
   // Build transformation centered at the surface position
-  Transform3 trafo(Transform3::Identity() * cfg.surfaceCfg.rotation);
-  trafo.translation() = cfg.surfaceCfg.position;
+  Vector3 centroid{0., 0., 0.};
+
+  for (const auto& surface : cfg.surfaces) {
+    centroid += surface->transform(gctx).translation();
+  }
+
+  centroid /= cfg.surfaces.size();
+
+  Transform3 trafo = Transform3::Identity();
+  trafo.translation() = centroid;
+  if (cfg.rotation) {
+    trafo.linear() = *cfg.rotation;
+  } else {
+    trafo.linear() = cfg.surfaces.front()->transform(gctx).rotation();
```

Comment:
Dumb question, but what makes the first surface special enough, that we can use its rotation? Calling ```front``` is also undefined behavior in case there are no ```surfaceCfg```s given or is this case excluded somewhere else?

Reply:
We have to make some choice about the rotation. The assumption is that this is a telescope-like geometry, where the sensors are located linearly but you're right that they might be rotated with respect to each other. We average the positions, but I don't think there's a good way to average the rotations accessible here.

Reply:
I see, thanks for clarifying. Maybe this should be commented then?

Reply:
That's a good idea. We should also add a check to see if `cfg.surfaceCfg.empty()`

Reply:
Should be added now.